src/utils/helpers.py

```diff
@@ -0,0 +1,25 @@
+import requests
+from PIL import Image
+from io import BytesIO
+from collections import Counter
+
+def get_dominant_color(image_url, threshold=200):
```

Comment:
Does this account for colors that are slightly different from each other? Like if there are 10 pixels with (1, 0, 0), 10 with (2, 0, 0), and 11 with (0, 1, 0), should the dominant color still be (0, 1, 0) even though there are more colors that are really similar to each other which make up more of the image?

Reply:
Oh, true! I've added a color quantization change to group similar colors together to account for slightly different shades, allowing the dominant color to better reflect the overall image composition.